tests/foundry-nft-f23/src/F1.sol

```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.10;
+
+import "./F2.sol";
+
+contract F1 {
+
+}
```

Comment:
In our current icf, this gets grouped into 0.8.25 context.
But in this PR, it can see that F2.sol is 0.8.21. Therefore F1.sol and F2.sol get grouped into 0.8.21 context